File: widgetssdk/src/main/java/com/glia/widgets/dialog/domain/SetOverlayPermissionRequestDialogShownUseCase.java
package com.glia.widgets.dialog.domain;

import com.glia.widgets.dialog.PermissionDialogManager;

public class SetOverlayPermissionRequestDialogShownUseCase {
    private final PermissionDialogManager permissionDialogManager;

    public SetOverlayPermissionRequestDialogShownUseCase(PermissionDialogManager permissionDialogManager) {
        this.permissionDialogManager = permissionDialogManager;
    }

    public void execute() {
        this.permissionDialogManager.setOverlayPermissionDialogShown();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/internal/dialog/domain/SetOverlayPermissionRequestDialogShownUseCase.kt
package com.glia.widgets.internal.dialog.domain

import com.glia.widgets.internal.dialog.PermissionDialogManager

internal interface SetOverlayPermissionRequestDialogShownUseCase {
    operator fun invoke()
}

internal class SetOverlayPermissionRequestDialogShownUseCaseImpl(private val permissionDialogManager: PermissionDialogManager) :
    SetOverlayPermissionRequestDialogShownUseCase {
    override fun invoke() = permissionDialogManager.setOverlayPermissionDialogShown()
}
